dashboard/test/models/user_test.rb

```diff
@@ -1616,6 +1616,51 @@ def track_progress(user_id, script_level, result, pairings: nil)
     assert_equal [], teacher.reload.permissions
   end
 
+  test 'grant admin permission logs to infrasecurity' do
+    teacher = create :teacher
+
+    teacher.stubs(:should_log?).returns(true)
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Granting UserPermission: environment: #{rack_env}, "\
+        "user ID: #{teacher.id}, "\
+        "email: #{teacher.email}, "\
+        "permission: ADMIN",
+        color: 'yellow'
+      ).
+      returns(true)
+
+    teacher.admin = true
+    teacher.save
+  end
+
+  test 'revoke admin permission logs to infrasecurity' do
+    admin_user = create :admin
+
+    admin_user.stubs(:should_log?).returns(true)
+    ChatClient.
+      expects(:message).
+      with('infra-security',
+        "Revoking UserPermission: environment: #{rack_env}, "\
+        "user ID: #{admin_user.id}, "\
+        "email: #{admin_user.email}, "\
+        "permission: ADMIN",
+        color: 'yellow'
+      ).
+      returns(true)
+
+    admin_user.admin = nil
+    admin_user.save
+  end
+
+  test 'grant admin permission does not log in test environment' do
+    ChatClient.expects(:message).never
+    teacher = create :teacher
+    teacher.admin = true
+    teacher.save
```

Comment:
Or use the [admin factory](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/factories/factories.rb#L60-L62) (which is equivalent to the above) but even more concise.

```ruby
admin = create :admin
```

Sorry for the slow delivery